### PR TITLE
fix clang build version on Ubuntu Artful

### PR DESCRIPTION
**Pull request by bukzor**
Ported from https://github.com/Linuxbrew/brew/pull/621

Before:

```
$ brew config | grep Clang:
Clang: 4.0 build 
```

After:

```
$ brew config | grep Clang:     
Clang: 4.0 build 401
```

### CODE REVIEW

File: Library/Homebrew/development_tools.rb [hunk 1]
```diff
@@ -74,7 +74,7 @@ def clang_version
     def clang_build_version
       @clang_build_version ||= begin
         if (path = locate("clang")) &&
-           build_version = `#{path} --version`[/clang-(\d{2,})/, 1]
+          build_version = `#{path} --version`[/clang(-|.*RELEASE_)(\d{2,})/, 2]
```
Comment (MikeMcQuaid) [inline Library/Homebrew/development_tools.rb]: This indentation change is incorrect; it's part of the `if` statement above. Also, what is present in the `.*` before `RELEASE`?

Comment (bukzor) [reply Library/Homebrew/development_tools.rb]: The string is `clang version 5.0.1 (tags/RELEASE_501/final)`.
According to the pastebins I was able to find, the "501" is what's found under osx.

Comment (MikeMcQuaid) [reply Library/Homebrew/development_tools.rb]: Cool, thanks. What about making it `tags/RELEASE_` instead of `.*`?

Comment (bukzor) [reply Library/Homebrew/development_tools.rb]: I was less certain that would continue to match, and I don't think the current pattern will over-match.

Comment (MikeMcQuaid) [reply Library/Homebrew/development_tools.rb]: @bukzor I'd rather we go for something more conservative for now and extended when needed. Thanks!

Comment (bukzor) [reply Library/Homebrew/development_tools.rb]: Can dooo!

File: Library/Homebrew/system_config.rb [hunk 2]
```diff
@@ -210,7 +210,7 @@ def dump_verbose_config(f = $stdout)
       f.puts "Homebrew Ruby: #{describe_homebrew_ruby}"
       f.puts "GCC-4.0: build #{gcc_4_0}" unless gcc_4_0.null?
       f.puts "GCC-4.2: build #{gcc_4_2}" unless gcc_4_2.null?
-      f.puts "Clang: #{clang.null? ? "N/A" : "#{clang} build #{clang_build}"}"
+      f.puts "Clang: #{clang.null? ? "N/A" : "#{clang} build #{clang_build.null? ? "???" : clang_build}"}"
```
Comment (bukzor) [inline Library/Homebrew/system_config.rb]: The intent is to make it more noticeable when the above regex doesn't match. I think it does a good job of that.
The output in that case looks like so:

```
Clang: 5.0 build ???
```

Comment (sjackman) [reply Library/Homebrew/system_config.rb]: I'd suggest `Clang: 5.0 build N/A`.

Comment (bukzor) [reply Library/Homebrew/system_config.rb]: "Not applicable" is the correct answer when a tool is not installed, which is its usage in the rest of this file, but the build number is indeed applicable when clang is installed, as it is here.

Comment (sjackman) [reply Library/Homebrew/system_config.rb]: Fair enough. I don't feel strongly.

**Comment by MikeMcQuaid** [comment]
@bukzor You can reproduce those errors locally with `brew style`.

File: Library/Homebrew/system_config.rb (hunk 2 above)
Comment (bukzor) [reply Library/Homebrew/system_config.rb]: <3

Comment (bukzor) [reply Library/Homebrew/system_config.rb]: I changed the "???" string to "(parse error)", which should be more clear.

**Comment by bukzor** [comment]
I'm not 100% happy with the result, but `brew style` is happy now.

**Comment by MikeMcQuaid** [comment]
Thanks again @bukzor!

**Comment by bukzor** [comment]
Thank you too @MikeMcQuaid. It's a pleasure to work with you all. You and the rest of the homebrew team are a role model for open-source upstreams, and collaborative software in general.

**Comment by MikeMcQuaid** [comment]
@bukzor Thanks for the kind words and taking the time to say them. We end up dealing with a lot of bile so comments like this are touching ❤️

**Pull request merged**